app/src/main/java/com/example/fast_pedals_frontend/bike/BikeScreen.kt

```diff
@@ -0,0 +1,387 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material.icons.automirrored.filled.DirectionsBike
+import androidx.compose.material.icons.filled.ContactMail
+import androidx.compose.material.icons.filled.LocationOn
+import androidx.compose.material.icons.filled.Money
+import androidx.compose.material.icons.filled.Favorite
+import androidx.compose.material.icons.filled.FavoriteBorder
+import androidx.compose.material.icons.filled.Person
+import androidx.compose.material.icons.filled.Phone
+import androidx.compose.material.icons.outlined.Info
+import androidx.compose.material.icons.outlined.Place
+import androidx.compose.material3.Button
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.example.fast_pedals_frontend.R
+import com.example.fast_pedals_frontend.bike.api.response.BikeResponse
+import com.example.fast_pedals_frontend.edit.SharedEditViewModel
+import com.example.fast_pedals_frontend.ui.theme.FastPedalsFrontEndTheme
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun BikeScreen(
+
+    bikeViewModel: BikeViewModel,
+    listingId: Long,
+    onBack: () -> Unit,
+    onEdit: (Long) -> Unit,
+    onDelete: () -> Unit,
+    sharedEditViewModel: SharedEditViewModel
+
+) {
+
+    val state by bikeViewModel.bikeState
+    val listing by bikeViewModel.listing.collectAsState()
+    val bike by bikeViewModel.bike.collectAsState()
+    val contactInfo by bikeViewModel.contactInfo.collectAsState()
+
+    bikeViewModel.getListing(listingId)
```

Comment:
this should be either in LaunchedEffect or some other effect. Read about it. Now its being executed after every recompsition.